### ISSUE TITLE
Drop support for python 3.4?

### ISSUE THREAD

**Issue by get9** [facebookincubator/dcrpm#21]
@fornellas is deprecating support for python 3.4 in TestSlide, the testing framework we use here. dcrpm still enforces compat with this version for $reasons. Can we drop support for it, especially since [it's been EOL'd upstream](https://devguide.python.org/devcycle/#end-of-life-branches)?

**Comment by davide125** [comment]
We need 3.4 to support CentOS 7, as psutil is only packaged for this version (as `python34-psutil`) and for python2 (as `python2-psutil`).

**Comment by jaymzh** [comment]
Indeed, there are users of this on c6 and c7 and thus need to continue to support them.

**Comment by malmond77** [comment]
Is it possible to pin the version of TestSlide to the last version that supports Python 3.4 when using Python3.4 ? @fornellas - is deprecating Python 3.4 support in TestSlide needed because of a won't fix issue upstream?

**Comment by malmond77** [comment]
Oh right that's what #22 is all about. I'm good with it, provided the tests pass.

**Comment by malmond77** [comment]
So #22 solves this problem another way. I believe this resolves this issue adequately.

**Comment by fornellas** [comment]
@malmond77 3.4 support was added specifically for dcrpm. It required adding inspect2 as a dependency to make it work, and this was causing [issues](https://github.com/JelleZijlstra/inspect2/issues/2) at some point. As I found out that 3.4 was EOL, I figured there was no reason to support it anymore, but apparently, RedHat's legacy here to haunt us...

I believe we can "if python 3.4 then don't do whatever inspect2 did" as a way of providing partial functionality to 3.4 (including disabling the specific test cases that break). Feel free to send a PR to TestSlide for this if you want, it should at least allow dcrpm to use newer versions of it.
